ccmbioinfo/lfs
# CARDS-2272: SlackNotificationsTask and NightlySlackNotifications should be moved into their own CARDS module

This Pull Request moves the `SlackNotificationsTask` and `NightlySlackNotifications` classes out of the `patient-portal` module and into their own new `slack-notifications` module thus allowing for the configuration of periodic performance metric alerting via Slack on projects that do not use the `patient-portal` module but still generate performance metrics (ie. _HERACLES_).

Testing Instructions
--------------------------

1. Build this (`CARDS-2272`) branch, including a Docker image, with `mvn clean install -Pdocker`.
2. In a new terminal window, enter the `Utilities/Development` directory and run `nodejs mock_slack.js`
3. In the `compose-cluster` directory, run:
    - `export SLACK_PERFORMANCE_URL=http://172.99.0.1:8000`
    - `export NIGHTLY_SLACK_NOTIFICATIONS_SCHEDULE='0 * * ? * *'`
    - `python3 generate_compose_yaml.py --dev_docker_image --oak_filesystem --composum --cards_project cards4prems --slack_notifications --subnet 172.99.0.0/16`
    - `docker-compose build && docker-compose up -d`
4. Monitor the terminal running `mock_slack.js` - performance update messages should be sent every minute
5. Stop the Docker Compose environment and clean up with `docker-compose down && docker-compose rm && docker volume prune -f && ./cleanup.sh`
6. In the root of the `cards` directory, run:
    - `export SLACK_PERFORMANCE_URL=http://localhost:8000`
    - `export NIGHTLY_SLACK_NOTIFICATIONS_SCHEDULE='0 * * ? * *'`
    - `./start_cards.sh --dev --project cards4prems --slack_notifications`
7. Monitor the terminal running `mock_slack.js` - performance update messages should be sent every minute
